src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            List<? extends Machine> machines = bulldozerProducer.get();
+            return new ArrayList<>(machines);
+        } else if (type == Truck.class) {
+            TruckProducer truckProducer = new TruckProducer();
+            List<? extends Machine> machines = truckProducer.get();
+            return new ArrayList<>(machines);
+        } else if (type == Excavator.class) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            List<? extends Machine> machines = excavatorProducer.get();
+            return new ArrayList<>(machines);
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, T value) {
+        int size = machines.size();
```

Comment:
I think you don`t need this variable, just write this in loop

Reply:
i agree